### DISCUSSION TITLE
-clip-path unable to target Photoshop path when the name includes extended characters

### DISCUSSION THREAD

**Discussion by FredrikEngquist** [ImageMagick/ImageMagick#5807] [Development]
When converting Photoshop images we have to address the clipping path by name since such images often contain many paths and the clipping path is not always guaranteed to be the first one.
This means I cannot use -clip or -clip-path "#1" but have to use the actual name -clip-path "path_name".

This works well in most situations (e.g. with the PS English default 'Path 1') but I have now encountered an issue where the clipping path name is 'Tracé 1', which is the Photoshop default name in French versions. I also have the same issue with images that seem to originate from Asia.

Photoshop 8BIM text seems to use it's own encoding (which may actually be MacRoman given it's origin) and I can extract the Clipping Path Name tag correctly using Exiftool with '-Photoshop:ClippingPathName -charset photoshop=MacRoman'. 

I use:
`magick 306629_03_2022_s.jpg[0] -background none -fill none -transparent none -alpha transparent -clip-path "Tracé 1" -alpha opaque -strip 306629_03_2022_s.png`
to create a transparent PNG from an original image with a clipping path.

The attached image '306629_03_2022_s.jpg' contains 2 paths; 'Path 1' and 'Tracé 1' where the latter is the clipping path.
![306629_03_2022_s](https://user-images.githubusercontent.com/69301299/204000647-1c53b6de-53ed-475e-8647-c058dc689a63.jpg)

**Comment by fmw42** [comment]
Post your input image so we can test with it.  Please also provide your IM version and platform/OS when asking questions.

**Comment by FredrikEngquist** [comment]
The image above is the image.
here is a zipped version.

Any version of IM will do.
7.1.0-52 behaves the same as any other v7.
I have tested on MacOS and Linux.

[306629_03_2022_s.jpg.zip](https://github.com/ImageMagick/ImageMagick/files/10094242/306629_03_2022_s.jpg.zip)

**Reply by fmw42** [reply]
Sorry, I thought your image was not the original.  I can confirm that your command does not work with that name using IM 7.1.0.52 on Mac OSX Monterey.

However, you can do this (by path number)

```
magick identify -quiet -format "%[8BIM:1999,2998:#2]" 306629_03_2022_s.jpg |\
magick -quiet 306629_03_2022_s.jpg +repage \
\( - -negate \) \
-alpha off -compose copy_opacity -composite \
306629_03_2022_s.jpg_clip.png

```

![306629_03_2022_s jpg_clip](https://user-images.githubusercontent.com/13458218/204029558-7e6332a1-5923-4ff4-a168-a868d363096e.png)


You can even anti-alias in the command if desired.

**Comment by urban-warrior** [comment]
ImageMagick supports UTF-8 encoding.  Instead put your `magick` command into `magick-utf8.sh` then type:

```
iconv -f UTF-8 -t MACINTOSH magick-utf8.sh > magick-mac.sh
sh magick-mac.sh
```

**Comment by snibgo** [comment]
Using "#2" as the name also works:
```
magick 306629_03_2022_s.jpg[0] -background none -fill none -transparent none -alpha transparent -clip-path "#2" -alpha opaque -strip 306629_03_2022_s.png
```

**Comment by FredrikEngquist** [comment]
I am aware that I can address the paths by number (-clip-path "#2"), but unfortunately I do not know of a way to determine the number.
The whole point is that the Photoshop clipping path is unknown since an image can contain any number of paths and any one of them can be the actual clipping path as set in Photoshop.
The only way I know to find it among all the paths in the image is to use Exiftool to extract the name of it. This does not give me the number, only the name.

Regarding the use of UTF-8, I do use UTF-8 for everything. The issue is that the encoding of the path names in the 8BIM profile is not UTF-8, but actually MacRoman as I suspected.
Exporting the 8BIM profile to a .8bimtext file confirms this as demonstrated in this clip:
```
---
8BIM#2000#Path 1="&#0;&#6;&#0;&#0;&#0;&#0;&#0;&#0;[...]
8BIM#2001#Tracé 1="&#0;&#6;&#0;&#0;&#0;&#0;&#0;&#0;[...]
8BIM#3000#OriginDataIRB="&#0;&#0;&#0;&#16;&#0;&#0;[...]
8BIM#2999="&#7;Trac&#142; 1&#0;&#0;&#0;&#0;&#0;&#1;"
---
```
'#142' is the MacRoman coding of é.

This means that if 'image.c/ClipImagePath' doesn't actively read the path names as MacRoman and translte it to UTF-8 when matching the input I give as the -clip-path value there is no chance of actually achieving a match.

Ultimately, I will either need a way to read the clipping path number from the image (and I don't know of any tool that does this) or I need a fix in IM that addresses the paths in the 8BIM profile using the correct text encoding.

**Comment by stefanhirt** [comment]
There still seems no solution to this, has anyone an alternative approach?

**Comment by urban-warrior** [comment]
Thank you for reporting the issue. We have successfully reproduced it and are actively working on a patch to resolve the issue. You can expect this patch to be merged into the [main GIT branch](https://github.com/ImageMagick/ImageMagick), later today. As part of our commitment to quality, this fix will also be included in the upcoming [beta releases of ImageMagick](https://imagemagick.org/archive/beta/) by tomorrow. Your patience and feedback are greatly appreciated.
